### PR TITLE
Avoid using default message for 401/403 responses, refactor utils.ts

### PR DESCRIPTION
**Pull request by chenhunghan**
- Avoid using default message for 401/403 responses
- Remove infamous `FALLBACK_HTTP_ERROR_CODE`
- refactor `utils.ts`, improve typing.

### CODE REVIEW

File: src/exceptions/utils.ts [hunk 1]
```diff
@@ -76,23 +62,34 @@ const DEFAULT_MAP: HTTPErrCodeExceptionMap = {
...
+      const mappedExceptionFn = exceptionsMap[httpStatusCode];
+
+      if (mappedExceptionFn) {
+        throw mappedExceptionFn(
+          await toPlatformErrorResponse(error?.response)
+        );
+      } else if (
+        httpStatusCode === 401 || httpStatusCode === 403
+      ) {
+        throw new LensSDKException(httpStatusCode, error.response?.data?.message, error);
```
Comment (panuhorsmalahti) [inline src/exceptions/utils.ts:87]: Shouldn't be throw ForbiddenException or UnauthorizedException? Otherwise this is a breaking change.

Comment (panuhorsmalahti) [reply src/exceptions/utils.ts:87]: Also, does `error.response?.data?.message` equal `error?.body.message`? If not, there is breaking change in src/UserService.ts.

Comment (chenhunghan) [reply src/exceptions/utils.ts:87]: `error.response?.data?.message equal error?.body.message` <== yes

Comment (chenhunghan) [reply src/exceptions/utils.ts]: This should be a major version anyway.

File: src/exceptions/utils.ts [hunk 2]
```diff
@@ -50,11 +40,7 @@ const toPlatformErrorResponse = async (response: undefined | unknown): Promise<P
  * };
  * ```
  */
-export type HTTPErrCodeExceptionMap<T = LensSDKException> = Partial<Record<HTTPErrorCode, (e?: PlatformErrorResponse) => T>>;
-
-const DEFAULT_MAP: HTTPErrCodeExceptionMap = {
-  401: e => new UnauthorizedException(e?.body?.message)
-};
+export type HTTPErrCodeExceptionMap<T = LensSDKException> = Partial<Record<number, (e?: PlatformErrorResponse) => T>>;
```
Comment (panuhorsmalahti) [inline src/exceptions/utils.ts:43]: Why `HTTPErrorCode` -> `number`?

Comment (chenhunghan) [reply src/exceptions/utils.ts:43]: I think we don't need to maintain a list of HTTP error code, the deleted one didn't cover all status codes in the SPEC, and axios type status code as `number`

**Review by panuhorsmalahti — APPROVED** [review]

**Pull request merged**